jib-core/src/main/java/com/google/cloud/tools/jib/frontend/JavaLayerConfigurations.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.frontend;
+
+import com.google.cloud.tools.jib.configuration.LayerConfiguration;
+import com.google.cloud.tools.jib.configuration.LayerConfigurations;
+import com.google.cloud.tools.jib.image.LayerEntry;
+import com.google.common.base.Preconditions;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/** Builds {@link LayerConfiguration}s for a Java application. */
+public class JavaLayerConfigurations extends LayerConfigurations {
```

Comment:
I'm not convinced that it's necessary that `JavaLayerConfigurations` extends `LayerConfigurations`. Personally, stylistically, I would avoid inheritance, and just have `JavaLayerConfigurations` implement some interface that defined a `LayerConfigurationCollection` type. And have `LayerConfigurations` be our default implementation for `jib-core` users. The lines like `return Preconditions.checkNotNull(getByLabel(RESOURCES_LAYER_LABEL)).getLayerEntries().get(0);` look kinda weird to me.

Reply:
Separating them out might also simplify the implementation of `LayerConfigurations` which seems (to me) to only have labeling to enable the `JavaLayerConfigurations` class?